modals/users.js

```diff
@@ -0,0 +1,30 @@
+const { v4 } = require('uuid');
+const md5 = require('md5');
+const { getUsers, setUsers } = require('./utils/storage.js');
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+function deleteUser(id) {
+    let users = getUsers();
+    const userToDelete = users.find(user => id === user.id);
+
+    if (userToDelete.role === 'admin') {
+        console.log(`You cant't delete "Admin". `);
```

Comment:
How would another function that uses `deleteUser` know that the user is admin and it was not deleted?

Reply:
After merging branches there will be createUser function, that will have user.role value

Reply:
The problem here is that `deleteUser` function doesn't communicate to it's parent function when it fails to delete a user.

Consider this use case:
```js
function parentFunction() {
   const userId = 123; // Imagine this is an admin user

   deleteUser(userId); // It has silently _failed_ to delete the user, because it's an admin user, but the parentFunction doesn't know it

   return "Success! The user was deleted."; // This is now incorrect - the user was NOT deleted, but the parentFunction doesn't know it
}
```

As you can see, `deleteUser` function doesn't signal to it's outer scope when it fails to delete user